pkg/agent/baker.go

```diff
@@ -220,7 +220,7 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
 			if profile.KubernetesConfig == nil {
 				return ""
 			}
-			return getDynamicKubeletConfigFileContent(profile.KubernetesConfig.KubeletConfig)
+			return GetDynamicKubeletConfigFileContent(profile.KubernetesConfig.KubeletConfig, profile.CustomKubeletConfig)
 		},
 		"IsDynamicKubeletEnabled": func() bool {
 			return IsDynamicKubeletEnabled(cs, config.EnableDynamicKubelet)
```

Comment:
nit: consider update the naming here now that we know it's not about dynamic or not

Reply:
This probably should be called something like useConfigFileForKubelet. I'll change this in another PR since it's a separate topic and code change is not trivial

Reply:
ok